src/PubmedId.php

```diff
@@ -3,9 +3,38 @@
 
 class PubmedId
 {
+    const REGEXP = <<<'EOT'
+{
+    (?<=
+        # Valid prefixes
+        ^                               # Start of the input
+        |
+        \s                              # Whitespace
+        |
+        ncbi\.nlm\.nih\.gov/pubmed/     # PubMed record page
+        |
+        ncbi\.nlm\.nih\.gov/m/pubmed/   # Mobile PubMed record page
+    )
+    0*          # Zero padding
+    (?!0)(\d+)  # Number (not starting with zero)
+    (?=
+        # Valid suffixes
+        $   # End of string
+        |
+        \s  # Whitespace
+        |
+        /   # Trailing slash
+        |
+        \?  # Query string
+        |
+        \#  # Fragment
+    )
+}xu
```

Comment:
This regexp is in contrast to @jbilbo's angle of attack in https://github.com/altmetric/identifiers/pull/20/ which is to have _two_ separate patterns.

At first (when doing the prefixes), I thought this might be simpler but the number of required suffixes might change my mind.

Reply:
The problem is the required suffixes for a zero-padded-integer pattern are different from the PubmedURL pattern, so the result code is not equivalent as having both separate, e.g. I don't think the zero-padded-integer should have `/`, `?` or `#` as acceptable suffixes, in the current regexp we'll extract `123456` from `123456#ABC` which looks wrong to me.

Reply:
That makes sense: I'll add some tests to ensure that doesn't happen and split the two patterns up.